optimade/server/config.py

```diff
@@ -137,42 +137,54 @@ def set_implementation_version(cls, v):
         res.update(v)
         return res
 
-    @validator("log_level")
-    def force_debug_log_level(cls, v, values):
-        """If `debug` is `True`, then force `log_level` to be `DEBUG` as well"""
-        from optimade.server.logger import CONSOLE_HANDLER
-
-        if values.get("debug", False):
-            v = LogLevel.DEBUG
-        CONSOLE_HANDLER.setLevel(v.value.upper())
-        return v
-
     @root_validator(pre=True)
-    def load_default_settings(cls, values):  # pylint: disable=no-self-argument
+    def load_settings(cls, values):  # pylint: disable=no-self-argument
         """
-        Loads settings from a root file if available and uses that as defaults in
-        place of built in defaults
+        Loads settings from a JSON config file, if available, and uses them in place
+        of the built-in defaults.
         """
-        from optimade.server.logger import LOGGER
-
         config_file_path = Path(values.get("config_file", DEFAULT_CONFIG_FILE_PATH))
 
         new_values = {}
 
-        if config_file_path.exists() and config_file_path.is_file():
-            LOGGER.debug("Found config file at: %s", config_file_path)
-            with open(config_file_path) as f:
-                new_values = json.load(f)
+        if config_file_path.is_file():
+            try:
+                with open(config_file_path) as f:
+                    new_values = json.load(f)
+            except json.JSONDecodeError as exc:
+                warnings.warn(
+                    f"Unable to parse config file {config_file_path} as JSON. Error: {exc}."
+                )
+
         else:
-            LOGGER.debug(  # pragma: no cover
-                "Did not find config file at: %s", config_file_path
-            )
+            if DEFAULT_CONFIG_FILE_PATH != str(config_file_path):
+                warnings.warn(
+                    f"Unable to find config file in requested location {config_file_path}, "
+                    "using the built-in default settings instead."
+                )
+            else:
+                warnings.warn(
+                    f"Unable to find config file in default location {DEFAULT_CONFIG_FILE_PATH}, "
+                    "using the built-in default settings instead."
+                )
+
+        if not new_values:
+            new_values["config_file"] = None
```

Comment:
Can be removed, right?
```suggestion
```

Reply:
Yeah I was being a bit over-defensive here, you're right